### PR TITLE
Mod 9701 fortmat mobile nav menu

### PR DESCRIPTION
**Pull request by chasls2**
**High level description:**

Created the mobile menu for the new nav bar dropdown

**JIRA Ticket:**
[DEV-9701](https://federal-spending-transparency.atlassian.net/browse/DEV-9701)

**Mockup:**
https://app.zeplin.io/project/6307b25775991866c016769c/dashboard?sid=6410856bee0f8d102180196a

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [x] Verified cross-browser compatibility: Chrome, Safari, Firefox, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
- [x] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [ ] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [ ] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [ ] Design review complete `if applicable`
- [ ] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/_scss/layouts/default/header/nav/_mobileHamburger.scss [hunk 1]
```diff
@@ -15,16 +15,14 @@
         height: rem(45);
         position: relative;
         width: rem(48);
-
```
Comment (brianpetway) [inline src/_scss/layouts/default/header/nav/_mobileHamburger.scss:18]: hover state for hamburger was removed; did design change it so there is no longer a hover state at all?

Comment (chasls2) [reply src/_scss/layouts/default/header/nav/_mobileHamburger.scss:18]: correct! let me find the message

File: src/_scss/layouts/default/header/nav/mobile/_nav.scss [hunk 2]
```diff
@@ -1,10 +1,17 @@
 .mobile-nav-content {
     ul.mobile-nav-content__list {
         @include unstyled-list;
-        margin-top: rem(50);
+        margin-top: rem(0);
+        // @media (min-width: $tablet-screen) {
```
Comment (brianpetway) [inline src/_scss/layouts/default/header/nav/mobile/_nav.scss:5]: can be removed?

Comment (chasls2) [reply src/_scss/layouts/default/header/nav/mobile/_nav.scss:5]: I think so, removing.

File: src/_scss/layouts/default/header/nav/_mobileHamburger.scss (hunk 1 above)
Comment (brianpetway) [reply src/_scss/layouts/default/header/nav/_mobileHamburger.scss:18]: it's ok, i believe you
just had to make sure

**Review by brianpetway — APPROVED** [review]

**Pull request merged**